MastodonSDK/Sources/MastodonUI/Helper/MediaLayoutHelper.swift

```diff
@@ -0,0 +1,314 @@
+//
+//  MediaLayoutHelper.swift
+//  
+//
+//  Created by Grishka on 25.03.2023.
+//
+
+import Foundation
+import MastodonSDK
+import CoreDataStack
+
+public struct MediaLayoutResult {
+	let width: Int
+	let height: Int
+	let columnSizes: [Int]
+	let rowSizes: [Int]
+	let tiles: [Tile]
+	
+	public struct Tile {
+		var colSpan: Int
+		let rowSpan: Int
+		var startCol: Int
+		let startRow: Int
+		var width: Int = 0
+	}
+}
+
+class MediaLayoutHelper {
+	static let maxWidth: Float = 1000
+	static let maxHeight: Float = 1777
+	static let minHeight: Float = 563
+	static let gap: Float = 1.5
+	static let maxRatio: Float = Float(maxWidth) / Float(maxHeight)
+	
+	public static func generateMediaLayout(attachments: [MastodonAttachment]) -> MediaLayoutResult? {
+		if attachments.count<2 {
+			return nil
+		}
+		
+		var ratios: [Float] = []
+		var allAreWide = true
+		var allAreSquare = true
+		for att in attachments {
+			let ratio: Float = Float(att.size.width/att.size.height)
+			if ratio<=1.2 {
```

Comment:
Can you add spaces around the operators? The rest of the app (and most Swift code except for the `..<` operator) do that and it improves readability a lot IMO
```suggestion
			if ratio <= 1.2 {
```